xarray/core/types.py

```diff
@@ -295,4 +295,5 @@ def copy(
 
 GroupKey = Any
 GroupIndex = Union[int, slice, list[int]]
-T_GroupIndices = list[GroupIndex]
+T_GroupIndices = tuple[GroupIndex, ...]
```

Comment:
I know that we are not very consistent with the naming, but I would still try to follow the rule that "T_" is reserved for TypeVars.